### ISSUE TITLE
Wrong prefix passed to providers

### ISSUE THREAD

**Issue by josa42** [atom/autocomplete-plus#855]
### Description

I am writing a Provider which works with `.hbs` files. In certain situations the prefix passed to the provider is `{` where I would expect an empty string.

### Example

In this situation the prefix passed to the Provider is `{`:

For a `.hbs` file with this content:

```
{{|}}
```
(`|` is the current cursor position)

Whereas if I start typing the `{` is correctly ignored. So for a file content like this the prefix is `xyz`.

```
{{xyz|}}
```

If more braces are added, they also get added to the prefix:

- `{{{|}}}` => Prefix: `{{`
- `{{{{|}}}}` => Prefix: `{{{`

### Steps to Reproduce

1. Add a package with a simple provider like this:
    
    ```
    export default {
      provideAutocomplete() {
        return {
          selector: '.text.html.mustache, .text.html.htmlbars',
          excludeLowerPriority: false,
          getSuggestions({ prefix }) {
            console.log('prefix', prefix)
            return [{ text: prefix }]
          }
        };
      }
    };
    ```

   Example Package: [autocomplete-example.zip](https://github.com/atom/autocomplete-plus/files/1023667/autocomplete-example.zip)

2. Open an `.hbs`. And add this content:
  
    ```
    {{|}}
    ```

    (`|` is the current cursor position)

2. Trigger `autocomplete-plus:activate`

**Expected behavior:** The prefix should be and empty string

**Actual behavior:** The prefix is `{`

### Versions

```
Atom    : 1.17.0
Electron: 1.3.15
Chrome  : 52.0.2743.82
Node    : 6.5.0
```
```
apm  1.17.0
npm  3.10.5
node 6.9.5 x64
python 2.7.13
git 2.13.0
```

```
OS: macOS 10.12
```

**Comment by jonyeezs** [comment]
I get the same issue as well

![image](https://user-images.githubusercontent.com/10862726/32166919-0fb2de28-bdb3-11e7-9363-9b0873f2e737.png)

```
getSuggestions({editor, bufferPosition, prefix}) {
//  bufferPosition: Point {row: 7, column: 9}
// prefix: " { "
```

```
this.selector = '.source.js, .source.ts, .source.tsx, .source.coffee';
```

**Comment by jonyeezs** [comment]
Any updates on this?

**Comment by leroix** [comment]
@jonyeezs I think it's possible that this will be fixed by #928 and/or #930, but I haven't had a chance to test yet.
